src/ngx_http_lua_balancer.c

```diff
@@ -681,9 +681,10 @@ ngx_http_lua_ffi_balancer_set_more_tries(ngx_http_request_t *r,
 
 #if (nginx_version >= 1007005)
     max_tries = r->upstream->conf->next_upstream_tries;
+    total = bp->total_tries + r->upstream->peer.tries - 1;
 
-    if (bp->total_tries + count > max_tries) {
-        count = max_tries - bp->total_tries;
+    if (max_tries && (total + count) > max_tries) {
```

Comment:
I think we'd better remove the parentheses around `total + count` since it's obvious that `+` has a higher priority than both `>` and `&&` :)